### ISSUE TITLE
Unable to connect

### ISSUE THREAD

**Issue by dsrtusr88** [jlesage/docker-firefox#141]
Not sure how to get appropriate logs for this.  Issue appeared after promoting from the last 2021 version.  I'm not seeing anything in the container dialogue that indicates a problem, but when I try to go to any website it just spins without ever connecting.

**Comment by jlesage** [comment]
So you do have access to Firefox itself ?

**Comment by dsrtusr88** [comment]
@jlesage that's correct, everything comes up clean, it just can't navigate anywhere off the default splash screen

**Comment by jlesage** [comment]
It looks like the container doesn't have access to internet... Did you create the container with the default bridge network ?

**Comment by dsrtusr88** [comment]
@jlesage I haven't changed any of the settings, I have it set to Host normally and I have tried with a bridge network with the same results.  This may be useful?

`[xvnc        ] Sun Oct 16 16:34:12 2022
[xvnc        ]  Connections: accepted: 10.0.3.1::51210
[xvnc        ]  VNCSConnST:  closing 10.0.3.1::51210: reading version failed: not an RFB
[xvnc        ]               client?
[xvnc        ]  EncodeManager: Framebuffer updates: 0
[xvnc        ]  EncodeManager:   Total: 0 rects, 0 pixels
[xvnc        ]  EncodeManager:          0 B (1:-nan ratio)
[xvnc        ]  Connections: closed: 10.0.3.1::51210
[xvnc        ]  ComparingUpdateTracker: 0 pixels in / 0 pixels out
[xvnc        ]  ComparingUpdateTracker: (1:-nan ratio)
[xvnc        ] Sun Oct 16 16:34:17 2022
[xvnc        ]  Connections: accepted: /tmp/vnc.sock
[xvnc        ]  SConnection: Client needs protocol version 3.8
[xvnc        ]  SConnection: Client requests security type None(1)
[xvnc        ]  VNCSConnST:  Server default pixel format depth 24 (32bpp) little-endian rgb888
[xvnc        ]  VNCSConnST:  Client pixel format depth 24 (32bpp) little-endian bgr888
[xvnc        ] Sun Oct 16 16:34:18 2022
[xvnc        ]  ComparingUpdateTracker: 0 pixels in / 0 pixels out
[xvnc        ]  ComparingUpdateTracker: (1:-nan ratio)`

**Comment by jlesage** [comment]
Are you able to access (from Firefox) something on your local network (e.g. your router's web interface)?

You can also try to login to the shell of the container (`docker exec -ti <container name> sh`) and see if you can ping something on the internet (e.g. `www.google.com`).

**Comment by dsrtusr88** [comment]
@jlesage from inside the container pings were returned from normal websites like google

**Comment by jlesage** [comment]
Ok, this is strange.  Maybe there is a setting inside Firefox that would cause the problem ?

Can you try to create a second container instance from scratch (without mapping to `/config`) to see if you have the same problem?

**Comment by dsrtusr88** [comment]
Okay, that worked but I'm getting an error on the following:
![image](https://user-images.githubusercontent.com/33256205/196575865-0c98b7ed-c7db-42a4-8eff-de4b57165ac5.png)
Of note, that error happened on the other config.

**Comment by jlesage** [comment]
Can you share the command you used to create the container ?

Also, which browser are you using the access the container ?

**Comment by dsrtusr88** [comment]
![image](https://user-images.githubusercontent.com/33256205/196581531-3244efc6-f7c8-447f-a5cd-59eda0eff9e8.png)
I just made a back up of my settings and will just build off a working config, but this is also odd that it's not loading with default values

**Comment by jlesage** [comment]
What do you mean by it's not loading with default values ?  Which default values are not working ?

**Comment by masicro** [comment]
I'am in China, but use the quick start to bulid contain ,it's not run. Why is it?

**Comment by dsrtusr88** [comment]
@jlesage in that quick build, normally every field populates values, it's odd those values in red are loading without entries.

**Comment by jlesage** [comment]
@masicro, please provide the docker run command you used with its output.  Also, note that the quick start is just an example and should be adapted to your particular situation.

**Comment by jlesage** [comment]
@dsrtusr88, not all environment variables need a value.  For some variables, like the ones you shown, no value is the default.
Does the creation of the container fail if these variables don't have a value ?

**Comment by masicro** [comment]
This is command is `docker run -d --name=firefox -p 5800:5800  --shm-size 2g  jlesage/firefox` and docker logs is that:

[init        ] container is starting...
[cont-env    ] loading container environment variables...
[cont-env    ] APP_NAME: loading...
[cont-env    ] APP_VERSION: loading...
[cont-env    ] DISPLAY: executing...
[cont-env    ] DISPLAY: terminated successfully.
[cont-env    ] DISPLAY: loading...
[cont-env    ] DOCKER_IMAGE_VERSION: loading...
[cont-env    ] GTK_THEME: executing...
[cont-env    ] GTK_THEME: terminated successfully.
[cont-env    ] GTK_THEME: loading...
[cont-env    ] HOME: loading...
[cont-env    ] QT_STYLE_OVERRIDE: executing...
[cont-env    ] QT_STYLE_OVERRIDE: terminated successfully.
[cont-env    ] QT_STYLE_OVERRIDE: loading...
[cont-env    ] TAKE_CONFIG_OWNERSHIP: loading...
[cont-env    ] XDG_CACHE_HOME: loading...
[cont-env    ] XDG_CONFIG_HOME: loading...
[cont-env    ] XDG_DATA_HOME: loading...
[cont-env    ] XDG_RUNTIME_DIR: loading...
[cont-env    ] container environment variables initialized.
[cont-secrets] loading container secrets...
[cont-secrets] container secrets loaded.
[cont-init   ] executing container initialization scripts...
[cont-init   ] 10-certs.sh: executing...
[cont-init   ] 10-certs.sh: terminated successfully.
[cont-init   ] 10-check-app-niceness.sh: executing...
[cont-init   ] 10-check-app-niceness.sh: terminated successfully.
[cont-init   ] 10-cjk-font.sh: executing...
[cont-init   ] 10-cjk-font.sh: terminated successfully.
[cont-init   ] 10-clean-logmonitor-states.sh: executing...
[cont-init   ] 10-clean-logmonitor-states.sh: terminated successfully.
[cont-init   ] 10-clean-tmp-dir.sh: executing...
[cont-init   ] 10-clean-tmp-dir.sh: terminated successfully.
[cont-init   ] 10-fontconfig-cache-dir.sh: executing...
[cont-init   ] 10-fontconfig-cache-dir.sh: terminated successfully.
[cont-init   ] 10-init-users.sh: executing...
[cont-init   ] 10-init-users.sh: terminated successfully.
[cont-init   ] 10-nginx.sh: executing...
[cont-init   ] 10-nginx.sh: terminated successfully.
[cont-init   ] 10-vnc-password.sh: executing...
[cont-init   ] 10-vnc-password.sh: terminated successfully.
[cont-init   ] 10-web-data.sh: executing...
[cont-init   ] 10-web-data.sh: terminated successfully.
[cont-init   ] 10-x11-unix.sh: executing...
[cont-init   ] 10-x11-unix.sh: terminated successfully.
[cont-init   ] 10-xdg-runtime-dir.sh: executing...
[cont-init   ] 10-xdg-runtime-dir.sh: terminated successfully.
[cont-init   ] 15-install-pkgs.sh: executing...
[cont-init   ] 15-install-pkgs.sh: terminated successfully.
[cont-init   ] 55-check-snd.sh: executing...
[cont-init   ] 55-check-snd.sh: sound not supported: device /dev/snd not exposed to the container.
[cont-init   ] 55-check-snd.sh: terminated successfully.
[cont-init   ] 55-firefox.sh: executing...
[cont-init   ] 55-firefox.sh: generating machine-id...
[cont-init   ] 55-firefox.sh: terminated successfully.
[cont-init   ] 56-firefox-set-prefs-from-env.sh: executing...
[cont-init   ] 56-firefox-set-prefs-from-env.sh: terminated successfully.
[cont-init   ] 85-take-config-ownership.sh: executing...
[cont-init   ] 85-take-config-ownership.sh: terminated successfully.
[cont-init   ] 89-info.sh: executing...
    ╭――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――╮
    │                                                                      │
    │ Application:          Firefox                                        │
    │ Application Version:  101.0.1-r0                                     │
    │ Docker Image Version: 22.10.3                                        │
    │                                                                      │
    ╰――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――╯
[cont-init   ] 89-info.sh: terminated successfully.
[cont-init   ] all container initialization scripts executed.
[init        ] giving control to process supervisor.
[supervisor  ] loading services...
[supervisor  ] loading service 'default'...
[supervisor  ] all services loaded.
[supervisor  ] starting services...
[supervisor  ] all services started.
[supervisor  ] all services exited, shutting down..
[finish      ] executing container finish scripts...
[finish      ] all container finish scripts executed.

**Comment by dsrtusr88** [comment]
@jlesage no, it still sets up the container.  I think I can close this comment, the only variable I isolated was my old config file that I've just moved on restoring a new config with back-up settings.  Ready to close the Issue.  I am still getting the red error box: 


Unexpected error encountered:
JSON.parse: unexpected character at line 1 column 1 of the JSON data

Not sure if it's related or a new Issue
